### PR TITLE
Team: allow lead to remove members

### LINKED ISSUE #280: Allow team lead to remove a member
marvel: https://marvelapp.com/5c70c7a/screen/55754080

Screen Shot 2019-10-04 at 11 12 44 PM

### PR DESCRIPTION
**Pull request by compnerd**
Enable team management for the lead.  Resolves #280!

### CODE REVIEW

**Comment by compnerd** [comment]
image
image
image

File: Steps4Impact/Settings/TeamSettingsViewController.swift [hunk 1]
```diff
@@ -92,6 +95,28 @@ extension TeamSettingsViewController: SettingsActionCellDelegate {
...
+  func removeTapped(context: Context?, button: UIButton) {
+    guard let context = context as? TeamMembersContext else { return }
+    switch context {
+    case .remove(let fbid, let name):
+      let alert: AlertViewController = AlertViewController()
+      alert.title = "Remove \(name)"
+      alert.body = "\(name) will be permanently removed from your team"
+      alert.add(AlertAction.cancel())
+      alert.add(AlertAction(title: "Remove", style: .destructive, shouldDismiss: false) {
+        AKFCausesService.leaveTeam(fbid: fbid) { (_) in
```
Comment (samisuteria) [inline Steps4Impact/Settings/TeamSettingsViewController.swift]: I'd add a case if the result comes back as a failure from the api

**Review by samisuteria — CHANGES_REQUESTED** [review]
Just handle failure case from api

**Comment by compnerd** [comment]
image

**Comment by compnerd** [comment]
image

File: Steps4Impact/Settings/TeamSettingsViewController.swift [hunk 2]
```diff
@@ -92,6 +95,41 @@ extension TeamSettingsViewController: SettingsActionCellDelegate {
...
+    switch context {
+    case .remove(let fbid, let name):
+      let alert: AlertViewController = AlertViewController()
+      alert.title = "Remove \(name)"
+      alert.body = "\(name) will be permanently removed from your team"
+      alert.add(AlertAction.cancel())
+      alert.add(AlertAction(title: "Remove", style: .destructive, shouldDismiss: false) {
+        AKFCausesService.leaveTeam(fbid: fbid) { (result) in
+          if !result.isSuccess {
+            alert.dismiss(animated: true, completion: nil)
```
Comment (samisuteria) [inline Steps4Impact/Settings/TeamSettingsViewController.swift]: I think you need to present the new alert in the completion of this alert

**Review by samisuteria — APPROVED** [review]
just a nit

**Review by samisuteria — APPROVED** [review]

**Pull request merged**